tests/flux-v2-image-automation.sh

```diff
@@ -56,7 +62,55 @@ for FILE_LOCATION in $(echo ${FILE_LOCATIONS}); do
                 echo "Non whitelisted pattern found in ImagePolicy: $IMAGE_POLICY it should be ^prod-[a-f0-9]+-(?P<ts>[0-9]+)" && exit 1
             fi
         done
+    done
+
+    ##############################################################################################################
+    # Print success if ALL image policies are clean
+    ##############################################################################################################
+    printf "\n\n########## Image Policy documents checked and passing ########## \n\n"
+
+    ##############################################################################################################
+    # This section compiles a list of files that contains `image:` and stores them in an array HELMRELEASES
+    # Only files that follow these rules will be added to the array:
+    # - Contains the `image:` string
+    # - Is HelmRelease type document
+    ##############################################################################################################
+
+    HELMRELEASES=()
+        for FILE in $(grep -lr "image:" $FILE_LOCATION | grep -Ev "$EXCLUSIONS" ); do
+            while read -r doc; do
+                if [ "$doc" == "HelmRelease" ]; then
+                    IFS=$'\n'
+                    HELMRELEASES+=("$FILE")
+                fi
+            done < <(yq eval '.kind' $FILE)
+        done
+
 
+    OUTPUTFILE="images.yaml"
+    DIRECTORIES="apps"
+
+    for dir in $DIRECTORIES; do
+        for EXCLUDED_PATH in "${HELMRELEASES[@]}"; do
+            if [[ $dir != $EXCLUDED_PATH* ]]; then
+                if ls "$dir" | grep -q -E 'kustomization\.ya?ml'; then
+                    kustomize build --load-restrictor LoadRestrictionsNone "$dir" 2>&1 | yq eval 'select(.kind == "HelmRelease" and (.spec.values.nodejs.image != null or .spec.values.java.image != null))' >> $OUTPUTFILE
```

Comment:
Your spec lookup needs to account for other image locations.
Look at mine and you'll see `spec.values.images` and `spec.values.*.images` those account for any wildcards and the top level values.images which the wildcard would not find.